### PR TITLE
Move key parsing

### PR DESCRIPTION
**Pull request by BeaudanBrown**
Parse the private key and generate the public key in main instead of channelEncryption
Set default key location

### CODE REVIEW

File: crypto/src/channel_encryption.cpp [hunk 1]
```diff
@@ -16,11 +15,8 @@ std::vector<uint8_t> hexToBytes(const std::string& hex) {
 }
 
 template <typename T>
-ChannelEncryption<T>::ChannelEncryption(const std::string& key_path) {
-    // Lokid uses ed25519
```
Comment (msgmaxim) [inline crypto/src/channel_encryption.cpp]: Is this comment still relevant?

Comment (BeaudanBrown) [reply crypto/src/channel_encryption.cpp]: This is deleted

Comment (msgmaxim) [reply crypto/src/channel_encryption.cpp]: Yeah, I'm asking if we should keep it instead

Comment (BeaudanBrown) [reply crypto/src/channel_encryption.cpp]: Ah right, it was there to explain the function converting from ed25519 to curve25519, guess couldn't hurt to put back?

**Pull request merged**